### DISCUSSION TITLE
v0.23.0

### DISCUSSION THREAD

**Discussion by sriramveeraghanta** [makeplane/plane#5765] [Announcements]
Here's everything that's new and better on Community.

#### Lossless edits and version history in Pages

Collaborate on Pages with your team without losing edits. Track changes in Pages with a version history going back 20 edits.

#### Favorites

Easily organize and access your favorite pages, projects, views, cycles, and modules under `Your Favourites` on the sidebar on the left.

## 👀 Improvements

 1. Guests can no longer create issues in projects or edit them. They can continue to create issues in Intake and see them through until they are closed.
 2. Viewers are now gone. Invite folks in as a Guest and toggle ↓ on to let them view all issues.
 3. Three new external API endpoints are added. You can now add members to your workspace and projects, and attach files to issues with external IDs and sources.
 4. `Inbox` is now `Intake` and `Notifications` is now `Inbox`. Simpler, we think.
 5. Pressing `/` now brings the focus to a page.
 6. The placeholder `Press / for commands` in issue comments is now `Add comments`.
 7. Updated empty state image for project views.
 8. `Load more` in layouts is now blue for better visibility.
 9. Private pages can now be made public even when locked. The reverse is also true.
10. `My Work` and `My Projects` are now `Your Work` and `Your Projects`. It's all yours, after all.
11. Only admins and creators of issues, cycles, modules, views, and pages can delete them.
12. The look and feel of your Profile is now better.
13. The drag experience in the page editor is now smoother.
14. Clicking cycle stats now takes you to the cycle details page with relevant filters applied.
15. Improved UI for the empty state of workspace notifications.
16. Images can now be viewed in an expanded view by clicking on the expand icon on the image.
17. Removed `Workspace setting`  form the sidebar now you can access it from the workspace dropdown.

## 🐛 Bug fixes

 1. Empty state for Views now has a button to add issues.
 2. Archived sub-issues will now show under Archives by default.
 3. Long project names are now truncated in the project dropdown in Analytics.
 4. Left navbar auto-collapses on small screens.
 5. Tooltips no longer overlap in the icon select dropdown in Pages.
 6. Fixed text overflow in the project select dropdown in `Export Issue`.
 7. Leaving a project now redirects you to the home screen without refreshing.
 8. CMD + Z now works fine in issue descriptions.
 9. A warning now appears if you forget to add a title to an issue.
10. Switching projects resets filters applied on intake issues.
11. Restoring or deleting issues updates the issue count in the archived issues header.
12. Headings 4, 5, and 6 in markdown now work as they should.
13. Fixed color inconsistencies in table headers for bold text.
14. Issue widgets for archived issues are no longer editable.
15. Changes to issue details now instantly reflect on the issue cards in the calendar layout.
16. Clicking `Mark as read` now updates the inbox count properly.
17. Mentioning a user no longer causes the focus to disappear.
18. Removing an issue from a cycle now removes it from the board immediately.
19. Long display names (over 20 characters) now trigger an error message.
20. Pressing Shift + Return in to-do lists correctly adds a new checkbox.
21. The image resizer outline now aligns correctly.
22. All details are now visible at the bottom of `My work` page.
23. Changing layouts or adding display options no longer hides the `Update View` button.
24. The progress bar now shows all state groups with their respective colors and state names on hover.
25. Workspace admins can now change the roles of other admins.
26. Exported issue file status updates automatically when processing is complete.
27. Toggling full-width mode in Pages is now faster in Safari.
28. Adding duplicate labels now shows a toast notification.
29. Adding an image to the `Create Issue` modal no longer pulls images from a page.
30. Cycle dates can no longer be changed for completed cycles.
31. Page links now work just fine.
32. Switching cycles or modules works even when one sidebar is already open.
33. Adding a parent to an issue from the peek-over view no longer reloads the screen.
34. Removed users’ avatars no longer appear on project cards.
35. Casing is now consistent throughout the platform.
36. Snoozed issues with past due dates no longer show the incorrect time remaining.
37. Deleting a cycle, module, view, or page no longer shows a false "failed to delete" toast notification.
38. The `Help` icon in the left navbar is better aligned.
39. Unauthorized users trying to delete issues now see the correct toast alert.
40. Password field no longer comes pre-filled during login.
41. Project members can no longer delete inbox issues.
42. Text overflow on hover is fixed for empty state comic buttons.
43. An error message now appears when you don't have permission to view a page.
44. Fixed double placeholders for list items.
45. After onboarding, users are now correctly redirected to the workspace.
46. Padding around the login screen is now consistent across all screen sizes.
47. Module progress charts now switch between points and issues for archived modules.

### **Fixed [setup.sh](http://setup.sh) for macOS support**

`setup.sh` now runs smoothly on macOS, so you can install and set up Plane on your Mac without the previous issues.

### **Self-hosted custom build**

You can now create custom containerized versions of your self-hosted instance to suit your specific self-hosting setup.

# 

<hr /><em>This discussion was created from the release <a href='https://github.com/makeplane/plane/releases/tag/v0.23.0'>v0.23.0</a>.</em>

**Comment by ja49619** [comment]
can someone tell me how to update?
I select ./setup.sh item 5. Upgrade and get this
![image](https://github.com/user-attachments/assets/e1b3ec11-8523-4221-b94c-dbf46682e1c9)

Ubuntu 24.04.1 LTS
Docker version 27.3.1

**Reply by ja49619** [reply]
updating docker compose solved the problem

```
sudo rm /usr/local/bin/docker-compose
sudo curl -L "https://github.com/docker/compose/releases/latest/download/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
sudo chmod +x /usr/local/bin/docker-compose
```

**Comment by AliDjango** [comment]
> You can now create custom containerized versions of your self-hosted instance to suit your specific self-hosting setup

Is there a guide or documentation  on how to do this?

**Comment by tomerh2001** [comment]
> ❗❗ IMPORTANT: When upgrading to v0.23, make sure you are using the latest version of Docker Compose. If you are on a custom Docker setup, you'll need to install RabbitMQ and provide the necessary env variables in app-env.

What are the env variables that I need to define? I can't find a `connection_url` like env variable that specifies how to access rabbitMQ...

What I added is this:
`compose.yml`:
```
  plane-mq:
    image: rabbitmq:3-management-alpine
    container_name: ${SLUG}-mq
    environment:
      RABBITMQ_DEFAULT_USER: ${RABBITMQ_USER}
      RABBITMQ_DEFAULT_PASS: ${RABBITMQ_PASSWORD}
      RABBITMQ_DEFAULT_VHOST: ${RABBITMQ_VHOST}
    volumes:
      - ${SERVICES_DATASET}/${SLUG}-rabbitmq:/var/lib/rabbitmq
    restart: unless-stopped
    logging:
      driver: json-file
      options:
        max-size: 10m
        max-file: 3
``` 

`.env`:
```
# RABBITMQ SETTINGS
RABBITMQ_HOST="plane-mq"
RABBITMQ_PORT="5672"
RABBITMQ_USER="plane"
RABBITMQ_PASSWORD="plane"
RABBITMQ_VHOST="plane"
```

But it doesn't work, what am I missing?

**Reply by pablohashescobar** [reply]
@tomerh2001, can you check if you are providing `AMQP_URL` to `api` , `worker` and `beat` containers.

**Comment by sriramveeraghanta** [comment]
Please use the latest version of [docker-compose](https://github.com/makeplane/plane/blob/master/deploy/selfhost/docker-compose.yml) while upgrading to v0.23.0.
